File: forcetracker333_ui/src/services/CommunicationService.js
import http from "../http-common";

class CommunicationService {
  getAllCommunications(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/communication/communications`, searchDTO);
  }

  get(communicationId) {
    return this.getRequest(`/communication/${communicationId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/communication?field=${matchData}`, null);
  }

  addCommunication(data) {
    return http.post("/communication/addCommunication", data);
  }

  update(data) {
  	return http.post("/communication/updateCommunication", data);
  }
  
  uploadImage(data,communicationId) {
  	return http.postForm("/communication/uploadImage/"+communicationId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new CommunicationService();

File: src/main/java/com/forcetracker333/service/CommunicationService.java
package com.forcetracker333.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.forcetracker333.domain.Communication;
import com.forcetracker333.dto.CommunicationDTO;
import com.forcetracker333.dto.CommunicationSearchDTO;
import com.forcetracker333.dto.CommunicationPageDTO;
import com.forcetracker333.dto.CommunicationConvertCriteriaDTO;
import com.forcetracker333.service.GenericService;
import com.forcetracker333.dto.common.RequestDTO;
import com.forcetracker333.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface CommunicationService extends GenericService<Communication, Integer> {

	List<Communication> findAll();

	ResultDTO addCommunication(CommunicationDTO communicationDTO, RequestDTO requestDTO);

	ResultDTO updateCommunication(CommunicationDTO communicationDTO, RequestDTO requestDTO);

    Page<Communication> getAllCommunications(Pageable pageable);

    Page<Communication> getAllCommunications(Specification<Communication> spec, Pageable pageable);

	ResponseEntity<CommunicationPageDTO> getCommunications(CommunicationSearchDTO communicationSearchDTO);
	
	List<CommunicationDTO> convertCommunicationsToCommunicationDTOs(List<Communication> communications, CommunicationConvertCriteriaDTO convertCriteria);

	CommunicationDTO getCommunicationDTOById(Integer communicationId);







}





